carbons/carbons.go

```diff
@@ -61,3 +61,65 @@ func DisableIQ(ctx context.Context, s *xmpp.Session, iq stanza.IQ) error {
 	)), &v)
 	return err
 }
+
+
+//DisableCarbon is an xmlstream.Transformer that inserts <private xmlns='urn:xmpp:carbons:2'/>
+//into an individual message, which in turn disables carbons for that message. If a message contains
+//<private xmlns='urn:xmpp:carbons:2'/> already then we will not add it. Other we will add it to everything
+//for now and sort it out later
+
+
+func DisableCarbon(r xml.TokenReader) xml.TokenReader{
+	var(
+		disabled bool
+		inner xml.TokenReader
+	)
+	return xmlstream.ReaderFunc(func() (xml.Token, error){
+		start:
+			if inner != nil{
+				token, err := inner.Token()
+				if err == io.EOF {
+					inner = nil
+					err = nil
+				}
+
+				return token, err
+			}
+
+			token, err := r.Token()
+			switch err {
+			case io.EOF:
+				if token == nil{
+					return nil, err
+				}
+				err = nil
+			case nil:
+			default:
+				return token, err
+			}
+
+			switch t := token.(type) {
+			case xml.StartElement:
+				switch{
+				//handling a message were carbon copy is already disabled
+				case t.Name.Local == "private" && t.Name.Space == NS:
+					disabled = true
+				//if we make it here,  we are disabling carbon for any message as long
+				case t.Name.Local == "message" :
```

Comment:
We also want to check the namespace here (and in the `EndElement` check below) just like in the receipts package.